### PR TITLE
feat: add addDeveloper function

### PR DESCRIPTION
**Pull request by abhishek-gkmit**
addDeveloper function validates the developer and if all the datatypes are correct and not empty then it will add the developer to the developers array

### CODE REVIEW

File: src/api.ts [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+  age: number;
+  isEmployed: boolean;
+  skills: Array<string>;
+  projects: Array<Project>;
+  experience: number;
+}
+
+const developers: Developer[] = [];
+
+function addDeveloper(developer: Developer): void {
```
Comment (adiTrivsme) [inline src/api.ts:19]: - remove unnecessary comments
- use the same format for error messages

File: src/api.ts [hunk 2]
```diff
@@ -0,0 +1,63 @@
+interface Project {
```
Comment (adiTrivsme) [inline src/api.ts:1]: move type declarations to a declaration file

File: src/api.ts [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+  // validating developer based on type and empty data
+  if (typeof developer.id !== 'number') {
+    throw Error("id is not correct");
+
+  } else if (typeof developer.name !== 'string' || developer.name === '') {
+    // type of name is not string or it's value is empty
+    // it's an error
+    throw Error("Error: name property is either empty or it's type is not correct");
+
+  } else if (typeof developer.age !== 'number' || developer.age < 14 /* boys who are under 14 are not developers */) {
```
Comment (adiTrivsme) [inline src/api.ts:29]: do not put comments under any code block

either put on top of statement or after the code block - `... developer.age < 14 ) { /**/`

File: src/api.ts [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+    throw Error("age is not right");
+
+  } else if (typeof developer.isEmployed !== 'boolean') {
+    throw Error("isEmployed is not boolean type");
+
+  } else if (!Array.isArray(developer.skills) || developer.skills.some((skill) => typeof skill !== 'string' || skill === '')
+  ) {
+    throw Error("skills are not right");
+
+  } else if (typeof developer.experience !== 'number' || developer.experience > developer.age - 14 /* developer's experience can not be greater than maximum experience */) {
```
Comment (adiTrivsme) [inline src/api.ts:39]: group conditions with brackets for better readability

`typeof developer.experience !== 'number' || (developer.experience > developer.age - 14)`

**Review by adiTrivsme — CHANGES_REQUESTED** [review]

File: src/api.ts [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+    return [...acc, skill];
+  }, []);
+
+  // validation successfull adding the developer to developers array
+  developers.push(developer);
+}
+
+function validateProjects(projects: Project[]): void {
+  // this function will validate the projects and throw an error if any project is invalid
+}
```
Comment (shelly19gkmit) [inline src/api.ts:62]: remove empty functions

File: src/api.ts (hunk 1 above)
Comment (abhishek-gkmit) [reply src/api.ts:19]: > remove unnecessary comments

removed these comments in the `addSkills` pull request, should I remove them in this branch too?

Comment (adiTrivsme) [reply src/api.ts:19]: no it's fine, 

but why are there duplicate commits in every PR?

**Pull request merged**